### ISSUE TITLE
[python] Add more user-level control for verbosity in `tiledbsoma.io`

### ISSUE THREAD

**Issue by bkmartinjr** [single-cell-data/TileDB-SOMA#390]
There is code in the core modules that are very verbose with their logging. I suspect the idea is to keep notebook users informed of import/export progress.  But many (most?) use cases are embedding the library into other apps & packages, which do not want logging around progress updates, etc.

Two suggestions:
* make this level of human-centered logging a local or global config (eg `from_anndata(verbose=True)`, so that the level of non-error-related verbosity can be reduced
* move this kind of logging out of the core modules (ie, only do it in the `io` package)

**Comment by johnkerl** [comment]
@bkmartinjr i very, very, very strongly want _minimal_ (more-than-nothing) ingest-logging _on by default_ in notebooks -- i made some effort to reduce the verbosity from what it was (`info` level is now shorter than `debug` -- thanks for the feedback that this is still "spammy") -- silent-by-default is just plain bad UX for operations which can take several minutes -- we need to be showing 'liveness by default' within notebook environments

another option might be to check some notebook-related environment variable at startup

**Comment by johnkerl** [comment]
note that for the immediate term one can do `tiledbsoma.logging.warn()`

**Comment by bkmartinjr** [comment]
is there a way to auto-magically detect "notebook" vs "not notebook"?  I will give it some thought, but I am not sure I have one. The possible alternative is an explicit config, _on_ by default, which an app (who wants less logging) can set to "off".

**Comment by johnkerl** [comment]
looks like we both had messages in flight at the same time -- `tiledbsoma.logging.warn()

`

**Comment by johnkerl** [comment]
@bkmartinjr is this resolved by #416?

**Comment by bkmartinjr** [comment]
>  is this resolved by #416?

No, I left the logging intact when I refactored the code.  We need to have a design discussion about this at some point before 1.0 (but I don't think it is urgent).

**Comment by johnkerl** [comment]
Cool @bkmartinjr -- I changed the title to _ IO module has spammy logging_
